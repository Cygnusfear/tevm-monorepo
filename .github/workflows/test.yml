on:
  push:
    branches:
      - main
  pull_request:
  merge_group:

name: Test

jobs:
  test:
    name: Test
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
        with:
          fetch-depth: 0
          submodules: recursive

      - name: "Setup"
        uses: ./.github/actions/setup

      - name: Run tests
        # using node because bun nx run-many is not doing coverage correctly in ci for some reason
        run: bun i && npx nx run-many --target=test:coverage
        env:
          NX_CLOUD_ACCESS_TOKEN: ${{ secrets.NX_CLOUD_ACCESS_TOKEN }}
          RPC_URL_MAINNET: ${{ secrets.RPC_URL_MAINNET }}

      - name: 'Report @evmts/core Coverage'
        if: always() # Also generate the report if tests are failing
        uses:  davelosert/vitest-coverage-report-action@v2
        with:
          name: "@evmts/core"
          working-directory: "./packages/core"
      - name: 'Report @evmts/ethers Coverage'
        if: always() # Also generate the report if tests are failing
        uses:  davelosert/vitest-coverage-report-action@v2
        with:
          name: "@evmts/ethers"
          working-directory: "./extensions/ethers"
      - name: 'Report @evmts/base Coverage'
        if: always() # Also generate the report if tests are failing
        uses:  davelosert/vitest-coverage-report-action@v2
        with:
          name: "@evmts/base"
          working-directory: "./bundler/packages/base"
      - name: 'Report @evmts/bun-plugin Coverage'
        if: always() # Also generate the report if tests are failing
        uses:  davelosert/vitest-coverage-report-action@v2
        with:
          name: "@evmts/bun-plugin"
          working-directory: "./bundler/packages/bun"
      - name: 'Report @evmts/esbuild-plugin Coverage'
        if: always() # Also generate the report if tests are failing
        uses:  davelosert/vitest-coverage-report-action@v2
        with:
          name: "@evmts/esbuild-plugin"
          working-directory: "./bundler/esbuild"
      - name: 'Report @evmts/ts-plugin Coverage'
        if: always() # Also generate the report if tests are failing
        uses:  davelosert/vitest-coverage-report-action@v2
        with:
          name: "@evmts/ts-plugin"
          working-directory: "./bundler/packages/ts-plugin"
      - name: 'Report @evmts/config Coverage'
        if: always() # Also generate the report if tests are failing
        uses:  davelosert/vitest-coverage-report-action@v2
        with:
          name: "@evmts/config"
          working-directory: "./bundler/packages/config"
      - name: 'Report @evmts/schemas Coverage'
        if: always() # Also generate the report if tests are failing
        uses:  davelosert/vitest-coverage-report-action@v2
        with:
          name: "@evmts/schemas"
          working-directory: "./packages/schemas"
      - name: 'Report @evmts/blockexplorer Coverage'
        if: always() # Also generate the report if tests are failing
        uses:  davelosert/vitest-coverage-report-action@v2
        with:
          name: "@evmts/blockexplorer"
          working-directory: "./packages/blockexplorer"
      - name: 'Report @evmts/runtime Coverage'
        if: always() # Also generate the report if tests are failing
        uses:  davelosert/vitest-coverage-report-action@v2
        with:
          name: "@evmts/runtime"
          working-directory: "./bundler/packages/runtime"
      - name: 'Report @evmts/solc Coverage'
        if: always() # Also generate the report if tests are failing
        uses:  davelosert/vitest-coverage-report-action@v2
        with:
          name: "@evmts/solc"
          working-directory: "./bundler/packages/solc"
      - name: 'Report @evmts/unplugin Coverage'
        if: always() # Also generate the report if tests are failing
        uses:  davelosert/vitest-coverage-report-action@v2
        with:
          name: "@evmts/unplugin"
          working-directory: "./bundler/packages/unplugin"
      - name: 'Report @evmts/resolutions Coverage'
        if: always() # Also generate the report if tests are failing
        uses:  davelosert/vitest-coverage-report-action@v2
        with:
          name: "@evmts/resolutions"
          working-directory: "./bundler/packages/resolutions"
      - name: 'Report @evmts/effect Coverage'
        if: always() # Also generate the report if tests are failing
        uses:  davelosert/vitest-coverage-report-action@v2
        with:
          name: "@evmts/effect"
          working-directory: "./packages/effect"
