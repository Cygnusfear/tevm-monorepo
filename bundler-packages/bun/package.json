{
	"name": "@tevm/bun-plugin",
	"version": "1.0.0-next.28",
	"private": false,
	"description": "A bun plugin for tevm",
	"keywords": [
		"bun",
		"plugin",
		"solidity",
		"sol",
		"tevm"
	],
	"repository": {
		"type": "git",
		"url": "https://github.com/evmts/tevm-monorepo.git",
		"directory": "bundler/plugin"
	},
	"license": "MIT",
	"contributors": [
		"Will Cory <willcory10@gmail.com>"
	],
	"type": "module",
	"exports": {
		".": {
			"import": {
				"types": "./types/index.d.ts",
				"default": "./dist/index.js"
			},
			"require": {
				"types": "./dist/index.d.cts",
				"default": "./dist/index.cjs"
			}
		},
		"./package.json": "./package.json"
	},
	"main": "dist/index.cjs",
	"module": "dist/index.js",
	"types": "types/index.d.ts",
	"files": [
		"dist",
		"src",
		"types"
	],
	"scripts": {
		"all": "bun run build && bun run lint && bun run format && bun run generate:docs && bun run test:coverage",
		"build": "nx run-many --targets=build:dist,build:types --projects=@tevm/bun-plugin ",
		"build:dist": "bunx tsup",
		"build:types": "bunx tsup --dts-only && bunx tsc --emitDeclarationOnly --declaration",
		"clean": "rm -rf node_modules && rm -rf artifacts && rm -rf dist && rm -rf cache",
		"format": "bunx rome format . --write",
		"format:check": "bunx rome format .",
		"generate:docs": "bunx typedoc",
		"lint": "bunx rome check . --apply-unsafe",
		"lint:check": "bunx rome check . --verbose",
		"lint:deps": "bunx depcheck",
		"lint:package": "bunx publint --strict && bunx attw --pack",
		"test": "bunx vitest --coverage",
		"test:coverage": "bunx vitest run --coverage",
		"test:run": "bunx vitest run",
		"test:ui": "bunx vitest --ui"
	},
	"dependencies": {
		"@tevm/base-bundler": "workspace:^",
		"@tevm/bundler-cache": "workspace:^",
		"@tevm/config": "workspace:^",
		"@tevm/solc": "workspace:1.0.0-next.28",
		"effect": "2.0.0-next.54",
		"solc": "0.8.23-fixed"
	},
	"devDependencies": {
		"@tevm/contract": "workspace:^",
		"@tevm/tsconfig": "workspace:^",
		"@tevm/tsupconfig": "workspace:^",
		"bun-types": "^1.0.24"
	},
	"publishConfig": {
		"access": "public"
	}
}
