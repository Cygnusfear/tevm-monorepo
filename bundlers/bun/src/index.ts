export * from './evmtsBunPlugin'
export * from '@evmts/config'
