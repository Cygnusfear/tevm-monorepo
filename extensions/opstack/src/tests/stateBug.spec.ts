// tevm@1.0.0-next.44
// @tevm/opstack@1.0.0-next.43
// Run with: pnpm ts-node bug-createTransaction-state/index.ts
import { expect, test } from 'bun:test';
import {
  createGasPriceOracle,
  createL1Block,
  type L1Client,
  createL1Client,
} from '../index.js';

/* --------------------------------- PREPARE -------------------------------- */
const DEPOSITOR_ACCOUNT = '0xDeaDDEaDDeAdDeAdDEAdDEaddeAddEAdDEAd0001';
const GasPriceOracle = createGasPriceOracle();
const L1Block = createL1Block();

const prepare = async () => {
  // Create client
  const client = createL1Client();

  // Set contracts
  await client.setAccount({
    address: L1Block.address,
    deployedBytecode: L1Block.deployedBytecode,
  });
  await client.setAccount({
    address: GasPriceOracle.address,
    deployedBytecode: GasPriceOracle.deployedBytecode,
  });

  return client;
};

/* ------------------------------- SET ECOTONE ------------------------------ */
const setEcotoneAndCheck = async (client: L1Client) => {
  try {
    // Set Ecotone
    const writeRes = await client.contract({
      ...GasPriceOracle.write.setEcotone(),
      caller: DEPOSITOR_ACCOUNT,
      createTransaction: true,
    });
    expect(writeRes).toMatchSnapshot()

    // Check if Ecotone is active
    const res =
      await client.contract({
        ...GasPriceOracle.read.isEcotone(),
      });
    expect(res).toMatchSnapshot()

    // Check again with createTransaction: true
    const res2 =
      await client.contract({
        ...GasPriceOracle.read.isEcotone(),
        createTransaction: true,
      });
    expect(res2).toMatchSnapshot()

    return {
      ecotoneActivatedCreateTransactionFalse: res.data,
      ecotoneActivatedcreateTransactionTrue: res2.data,
    };
  } catch (err) {
    console.error('Error:', err);
    return {};
  }
};

/* ---------------------------------- MAIN ---------------------------------- */
const run = async () => {
  const client = await prepare();

  // Set Ecotone and get both results
  const {
    ecotoneActivatedCreateTransactionFalse,
    ecotoneActivatedcreateTransactionTrue,
  } = await setEcotoneAndCheck(client);
  expect(ecotoneActivatedcreateTransactionTrue).toBe(true);
  expect(ecotoneActivatedCreateTransactionFalse).toBe(true);
};

test('The optimism memory clients that are running in normal mode with a normal state manager should successfully deep clone state when running a call with `createTransaction=false`', run);
