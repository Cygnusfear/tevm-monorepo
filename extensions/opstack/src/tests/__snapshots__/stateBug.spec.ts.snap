// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`The optimism memory clients that are running in normal mode with a normal state manager should successfully deep clone state when running a call with `createTransaction=false` 1`] = `
{
  "createdAddresses":   Set {},
  "data": undefined,
  "executionGasUsed": 25588n,
  "gas": 16751627n,
  "logs": [],
  "rawData": "0x",
  "selfdestruct":   Set {},
}
`;

exports[`The optimism memory clients that are running in normal mode with a normal state manager should successfully deep clone state when running a call with `createTransaction=false` 2`] = `
{
  "createdAddresses":   Set {},
  "data": true,
  "executionGasUsed": 2377n,
  "gas": 16774838n,
  "logs": [],
  "rawData": "0x0000000000000000000000000000000000000000000000000000000000000001",
  "selfdestruct":   Set {},
}
`;

exports[`The optimism memory clients that are running in normal mode with a normal state manager should successfully deep clone state when running a call with `createTransaction=false` 3`] = `
{
  "createdAddresses":   Set {},
  "data": true,
  "executionGasUsed": 377n,
  "gas": 16776838n,
  "logs": [],
  "rawData": "0x0000000000000000000000000000000000000000000000000000000000000001",
  "selfdestruct":   Set {},
}
`;
